core/divelist.c

```diff
@@ -1315,24 +1332,17 @@ void process_dives(bool is_imported, bool prefer_imported)
 		delete_single_dive(i + 1);
 		// keep the id or the first dive for the merged dive
 		merged->id = id;
-
-		/* this means the table was changed */
-		did_merge = true;
 	}
+
 	/* make sure no dives are still marked as downloaded */
 	for (i = 1; i < dive_table.nr; i++)
```

Comment:
BTW: Is the `i = 1` here correct?
After all we *can* import dives into an empty log, isn't it..?

Reply:
But since we access dive_table.dives[i-1] we really don't want to start at 0 :-)
What this loop does is that it walks the whole dive table and compares "neighboring" dives to check if they should be merged.

Reply:
I think we're talking about different loops.

Note that if we import into an empty log, then `preexisting==0` and with
```
        for (i = preexisting; i < dive_table.nr; i++)
                dive_table.dives[i]->downloaded = true;
```
the `downloaded` flag of all dives will be set.

But with
```
        for (i = 1; i < dive_table.nr; i++)
                dive_table.dives[i]->downloaded = false;
```
it will not be reset for the first dive.

Or perhaps I'm missing something obvious.

Reply:
Yeah, I think that second for-loop should start from 0.